### PR TITLE
fix(mongodb): fix the onboarding cli when mongodb does't support js (#311)

### PR DESCRIPTION
**Pull request by Scra3**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

Link to CU-1p2cc76

### CODE REVIEW

**Comment by Scra3** [comment]
[TEST CASE SETUP]
**Mongodb instance without js support :**
host: cluster0.b6bd0.mongodb.net
user name: forest
mdp: forestalban
db: sample_analytics

File: src/services/schema/update/analyzer/mongo-collections-analyzer.js [hunk 1]
```diff
@@ -85,97 +94,235 @@ function mapCollection() {
...
+    getMongooseEmbeddedSchema,
+    getMongooseSchema,
+    haveSameEmbeddedType,
+    hasEmbeddedTypes,
+    mergeAnalyzedSchemas,
+    isSystemCollection,
+    getCollectionName,
+    mapCollection,
+    reduceCollection,
+  };
```
Comment (slimee) [inline src/services/schema/update/analyzer/mongo-collections-analyzer.js:324]: This sounds like an hack, there should be only one sealed context. Is there any reasons for example mongo side script?

Comment (slimee) [reply src/services/schema/update/analyzer/mongo-collections-analyzer.js:324]: all functions `detectReferences`, `applyReferences` etc. should be added to the context. Then  `MongoCollectionsAnalyzer` should be added too to the context.

**Review by slimee — COMMENTED** [review]
Quick read, the review not finished

File: src/services/schema/update/analyzer/mongo-collections-analyzer.js (hunk 1 above)
Comment (Scra3) [reply src/services/schema/update/analyzer/mongo-collections-analyzer.js]: ✅

**Review by slimee — APPROVED** [review]
nice job!

I've put some comments, many are just infos.

I let you do as you wish to change things.

File: src/context/command-projects-common-plan.js [hunk 2]
```diff
@@ -1,6 +1,64 @@
...
+  hasEmbeddedTypes,
+  mergeAnalyzedSchemas,
+} = require('../services/schema/update/analyzer/mongo-embedded-analyzer');
+
+const {
+  MongoCollectionsAnalyzer, mapCollection, reduceCollection,
+} = require('../services/schema/update/analyzer/mongo-collections-analyzer');
+
+function makeProgressBar(message, total) {
+  return new ProgressBar(`${message} [:bar] :percent`, { total, width: 50 });
```
[the remark this answers is not in the document]
Comment (Scra3) [reply src/context/command-projects-common-plan.js:38]: I will do an other PR to cover this. There are a lot of changes in this PR.

File: src/services/schema/update/analyzer/database-analyzer.js [hunk 3]
```diff
@@ -36,7 +36,10 @@ module.exports = class DatabaseAnalyzer {
   async analyze(databaseConnection, config, allowWarning) {
     let analyze;
     if (config.dbDialect === 'mongodb') {
-      analyze = this.mongoAnalyzer;
+      analyze = (databaseConnectionArg, configArg, allowWarningArg) =>
+        this.mongoAnalyzer.analyzeMongoCollections(
+          databaseConnectionArg, configArg, allowWarningArg,
+        );
```
[the remark this answers is not in the document]
Comment (Scra3) [reply src/services/schema/update/analyzer/database-analyzer.js:42]: yeah good idea !

File: test/services/analyzer/mongo-collections-analyzer.test.js [hunk 4]
```diff
@@ -1,18 +1,1131 @@
...
+  });
+
+  describe('mapCollection', () => {
+    describe('when the key is an ObjectId', () => {
+      describe('when the key is not "_id"', () => {
+        it('should emit the key with the good type', () => {
+          expect.assertions(2);
+
+          const emit = jest.fn();
+          const store = {};
```
[the remark this answers is not in the document]
Comment (Scra3) [reply test/services/analyzer/mongo-collections-analyzer.test.js:34]: the test does not compare the object reference ? if yes, we can not have a false positive.

**Pull request merged**